### PR TITLE
Refactor: Chat API objects

### PR DESCRIPTION
**Pull request by DanVanAtta**
- Updates chat api to use consolidated value objects.
- Forward looking, moves chat objects to http-client,
  prepping for usage with http-chat

 

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[x] Code Cleanup or refactor
[] Configuration Change
[] Bug fix:  
[] Other:   

## Testing

[x] No manual testing done
[] Manually tested

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/chat/Chat.java [hunk 1]
```diff
@@ -96,7 +98,7 @@ public void participantRemoved(final PlayerName playerName) {
   @Override
   public void slappedBy(final PlayerName from) {
     final String message = "You were slapped by " + from;
-    chatHistory.add(new ChatMessage(message, from));
+    chatHistory.add(new ChatMessage(from, message));
     chatMessageListeners.forEach(listener -> listener.slapped(message, from));
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/chat/Chat.java]: Why is the new ChatMessage object not passed to the listener?

File: game-core/src/main/java/games/strategy/engine/chat/ChatMessagePanel.java [hunk 2]
```diff
@@ -224,10 +224,10 @@ private void cleanupKeyMap() {
     nextMessageKeymap.remove(KeyStroke.getKeyStroke(KeyEvent.VK_DOWN, 0, false));
   }
 
-  /** thread safe. */
   @Override
-  public void messageReceived(final String message, final PlayerName from) {
-    addMessageWithSound(message, from, SoundPath.CLIP_CHAT_MESSAGE);
+  public void messageReceived(final ChatMessage chatMessage) {
+    addMessageWithSound(
+        chatMessage.getMessage(), chatMessage.getFrom(), SoundPath.CLIP_CHAT_MESSAGE);
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/chat/ChatMessagePanel.java]: Here and in line 115:
Why isn't ChatMessage passed there as well?

File: game-core/src/main/java/games/strategy/engine/chat/Chat.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/chat/Chat.java]: There is a 'slapped' event object that is coming along that should be part of the slapped API. It's a bit weird though as slap is handled essentially as a special cased message and not its own event. TBD exactly how to best handle this.

File: game-core/src/main/java/games/strategy/engine/chat/ChatMessagePanel.java (hunk 2 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/chat/ChatMessagePanel.java]: Same reasoning as previous, essentially a slap event will be coming along, not sure yet how to reconcile that with the 'addMessageWithSound'

**Pull request merged**